sdk/src/azure/platform/az_posix.c

```diff
@@ -26,3 +57,165 @@ AZ_NODISCARD az_result az_platform_sleep_msec(int32_t milliseconds)
+{
+  _az_PRECONDITION_NOT_NULL(timer_handle);
+
+  timer_handle->_internal.trigger.it_value.tv_sec = milliseconds / _az_TIME_MILLISECONDS_PER_SECOND;
+  timer_handle->_internal.trigger.it_value.tv_nsec
+      = (milliseconds % _az_TIME_MILLISECONDS_PER_SECOND) * _az_TIME_NANOSECONDS_PER_MILLISECOND;
+
+  if (0
+      != timer_settime(timer_handle->_internal.timerid, 0, &timer_handle->_internal.trigger, NULL))
+  {
+    if (EFAULT == errno)
+      return AZ_ERROR_ARG;
+    else if (EINVAL == errno)
+      return AZ_ERROR_ARG;
+    else
+      return AZ_ERROR_ARG;
+  }
+
+  return AZ_OK;
+}
+
+AZ_NODISCARD az_result az_platform_timer_destroy(_az_platform_timer* timer_handle)
+{
+  _az_PRECONDITION_NOT_NULL(timer_handle);
+
+  if (0 != timer_delete(timer_handle->_internal.timerid))
+  {
+    return AZ_ERROR_ARG;
+  }
+
+  return AZ_OK;
+}
+
+AZ_NODISCARD az_result az_platform_mutex_init(az_platform_mutex* mutex_handle)
+{
+  _az_PRECONDITION_NOT_NULL(mutex_handle);
+
+  pthread_mutexattr_t attr;
+  pthread_mutexattr_init(&attr);
+  pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE);
+
+  int mutex_init_result = pthread_mutex_init(mutex_handle, &attr);
+
+  if (EAGAIN == mutex_init_result)
+    return AZ_ERROR_RESOURCE_UNAVAILABLE;
+  else if (ENOMEM == mutex_init_result)
+    return AZ_ERROR_OUT_OF_MEMORY;
+  else if (EPERM == mutex_init_result)
+    return AZ_ERROR_PERMISSION;
+  else if (EBUSY == mutex_init_result)
+    return AZ_ERROR_REINITIALIZATION;
+  else if (EINVAL == mutex_init_result)
+    return AZ_ERROR_ARG;
+  else if (0 != mutex_init_result)
+    return AZ_ERROR_ARG;
+
+  return AZ_OK;
+}
+
+AZ_NODISCARD az_result az_platform_mutex_acquire(az_platform_mutex* mutex_handle)
+{
+  _az_PRECONDITION_NOT_NULL(mutex_handle);
+
+  int mutex_lock_result = pthread_mutex_lock(mutex_handle);
+
+  if (EINVAL == mutex_lock_result)
+    return AZ_ERROR_ARG;
+  else if (EBUSY == mutex_lock_result)
+    return AZ_ERROR_MUTEX_BUSY;
+  else if (EAGAIN == mutex_lock_result)
+    return AZ_ERROR_MUTEX_MAX_RECURSIVE_LOCKS;
+  else if (EDEADLK == mutex_lock_result)
+    return AZ_ERROR_DEADLOCK;
+  else if (0 != mutex_lock_result)
+    return AZ_ERROR_ARG;
+
+  return AZ_OK;
+}
+
+AZ_NODISCARD az_result az_platform_mutex_release(az_platform_mutex* mutex_handle)
+{
+  _az_PRECONDITION_NOT_NULL(mutex_handle);
+
+  int mutex_unlock_result = pthread_mutex_unlock(mutex_handle);
+
+  if (EINVAL == mutex_unlock_result)
+    return AZ_ERROR_ARG;
+  else if (EAGAIN == mutex_unlock_result)
+    return AZ_ERROR_MUTEX_MAX_RECURSIVE_LOCKS;
+  else if (EPERM == mutex_unlock_result)
+    return AZ_ERROR_PERMISSION;
+  else if (0 != mutex_unlock_result)
+    return AZ_ERROR_ARG;
+
+  return AZ_OK;
+}
+
+AZ_NODISCARD az_result az_platform_mutex_destroy(az_platform_mutex* mutex_handle)
+{
+  _az_PRECONDITION_NOT_NULL(mutex_handle);
+
+  int mutex_destroy_result = pthread_mutex_destroy(mutex_handle);
+
+  if (EBUSY == mutex_destroy_result)
```

Comment:
Please remove unused adapter code. We may add actionable error codes later, as needed.